db/order_puppies.js

```diff
@@ -0,0 +1,92 @@
+const client = require("./client");
+
+async function addToOrderPuppies({
+    userId,
+    puppyId
+}) {
+    try{
+        const { rows: [order_puppy] } = await client.query(`
+            INSERT INTO order_puppies("userId", "puppyId")
+            VALUES ($1, $2)
+            ON CONFLICT ("userId", "puppyId") DO NOTHING
+            RETURNING *;
+        `, [userId, puppyId])
+
+        return order_puppy;
+    } catch (error) {
+        console.error(error)
+    }
+}
+
+async function getOrderPuppyById(id) {
+    try{
+        const { rows: [ order_puppy ] } = await client.query(`
+            SELECT *
+            FROM order_puppies
+            WHERE id=$1;
+        `, [id])
+
+        return order_puppy;
+    } catch (error) {
+        console.error(error)
+    }
+}
+  
+async function getOrderPuppiesByOrder({ id }) {
+    try{
+        const { rows } = await client.query(`
+            SELECT *
+            FROM order_puppies
+            WHERE "orderId"=$1;
+        `, [id])
+
+        return rows;
+    } catch (error) {
+        console.error(error)
+    }
+}
+  
+async function updateOrderPuppy({ id, ...fields }) {
```

Comment:
I don't think we need an updateOrderPuppy function since it's just a table with reference ids so there shouldn't ever be a reason to update the entries. We'll just be adding and removing them.